bundle/config/mutator/python/apply_python_output.go

```diff
@@ -0,0 +1,212 @@
+package python
+
+import (
+	"fmt"
+
+	"github.com/databricks/cli/bundle/config/mutator"
+	"github.com/databricks/cli/libs/dyn"
+	"github.com/databricks/cli/libs/dyn/merge"
+)
+
+// applyPythonOutputResult contains which resources where added, updated, or deleted by Python mutator.
+type applyPythonOutputResult struct {
+	AddedResources   mutator.ResourceKeySet
+	UpdatedResources mutator.ResourceKeySet
+	DeletedResources mutator.ResourceKeySet
+}
+
+// applyPythonOutput applies output of Python mutator to bundle configuration before Python mutator.
+//
+// It records applyPythonOutputResult containing which resources where added, updated, or deleted
+// by Python mutator.
+//
+// Return value is equivalent to output except for:
+// - if property is unchanged in output, it's original location will be preserved
+// - if empty sequence/mapping is deleted in output, it's original value will be preserved
```

Comment:
s/it's/its/g

Regarding the empty sequence/mapping -- I recall this is due to `None` ergonomics in Python. Could you include a reference here that explains the full rationale?